powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ApplicationConfigServiceBehavior.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.service.behavior.tasks;
+
+import com.wultra.security.powerauth.client.model.entity.ApplicationConfigurationItem;
+import com.wultra.security.powerauth.client.model.request.CreateApplicationConfigRequest;
+import com.wultra.security.powerauth.client.model.request.GetApplicationConfigRequest;
+import com.wultra.security.powerauth.client.model.request.RemoveApplicationConfigRequest;
+import com.wultra.security.powerauth.client.model.response.*;
+import io.getlime.core.rest.model.base.response.Response;
+import io.getlime.security.powerauth.app.server.database.RepositoryCatalogue;
+import io.getlime.security.powerauth.app.server.database.model.entity.ApplicationConfigEntity;
+import io.getlime.security.powerauth.app.server.database.model.entity.ApplicationEntity;
+import io.getlime.security.powerauth.app.server.database.repository.ApplicationConfigRepository;
+import io.getlime.security.powerauth.app.server.database.repository.ApplicationRepository;
+import io.getlime.security.powerauth.app.server.service.exceptions.GenericServiceException;
+import io.getlime.security.powerauth.app.server.service.i18n.LocalizationProvider;
+import io.getlime.security.powerauth.app.server.service.model.ServiceError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Behavior class implementing management of application configuration.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Component("applicationConfigServiceBehavior")
```

Comment:
Is the value really needed? This name should be the same by default. Even though it is initialized to `new ArrayList<>()`.

Reply:
We can remove the qualifier, sure.